### PR TITLE
CO-1954 confirmation page after step 2 is working

### PR DESCRIPTION
**Pull request by quentingigon**

### CODE REVIEW

File: sms_sponsorship/controllers/main.py [hunk 1]
```diff
@@ -124,6 +124,16 @@ def sms_change_child(self, child_request_id):
 
         sms_child_request.change_child()
 
+    @route('/sms_registration_confirmation/<model('
```
Comment (ecino) [inline sms_sponsorship/controllers/main.py]: Route should be ''/sms_sponsorship/step2/<model("recurring.contract"):sponsorship>/confirm" for consistency

File: sms_sponsorship/forms/sms_sponsorship_registration_form.py [hunk 2]
```diff
@@ -209,7 +209,24 @@ def form_after_create_or_update(self, values, extra_values):
                 _logger.error("Activate sponsorship is not yet implemented")
 
             # update sms request
-            sms_request.complete_step2()
+            self.env['sms.child.request'].sudo()\
```
Comment (ecino) [inline sms_sponsorship/forms/sms_sponsorship_registration_form.py]: This is not good, you reverted some of the code that I refactored. Please rebase your branch onto devel and verify that you don't revert the code.

File: sms_sponsorship/templates/sms_registration_confirmation_template.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
```
Comment (ecino) [inline sms_sponsorship/templates/sms_registration_confirmation_template.xml]: Should be "odoo"

File: sms_sponsorship/templates/sms_registration_confirmation_template.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    <data>
```
Comment (ecino) [inline sms_sponsorship/templates/sms_registration_confirmation_template.xml]: data tag is useless

File: sms_sponsorship/templates/sms_registration_confirmation_template.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        <template id="sms_registration_confirmation" name="SMS registration confirmation">
+            <t t-call="website.layout">
+                <t t-set="title">Thank you for registration!</t>
+                <div class="oe_structure donation_success">
+                    <div class="thank_you_container">
+                        <div class="thank_you_image_container">
+                            <img class="img-responsive" style="width: 100%" src="/sms_sponsorship/static/src/img/compassion_thankyou_donation.jpg"/>
```
Comment (ecino) [inline sms_sponsorship/templates/sms_registration_confirmation_template.xml]: You forgot to commit the image. I think as well class img-responsive should be defined somewhere.

File: sms_sponsorship/templates/sms_registration_confirmation_template.xml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+                        </div>
+                        <div class="thank_you_text_container">
```
Comment (ecino) [inline sms_sponsorship/templates/sms_registration_confirmation_template.xml]: The class should be defined in the module

File: sms_sponsorship/templates/sms_registration_confirmation_template.xml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+                            <h1>Registration successful !</h1>
+                            <p class="quote">
```
Comment (ecino) [inline sms_sponsorship/templates/sms_registration_confirmation_template.xml]: The class should be defined

**Review by ecino — CHANGES_REQUESTED** [review]

File: sms_sponsorship/templates/sms_registration_confirmation_template.xml (hunk 5 above)
Comment (quentingigon) [reply sms_sponsorship/templates/sms_registration_confirmation_template.xml]: Image is committed

Comment (quentingigon) [reply sms_sponsorship/templates/sms_registration_confirmation_template.xml]: img-responsive is defined in odoo_ext/static/style

File: sms_sponsorship/static/src/css/custom.css [hunk 8]
```diff
@@ -0,0 +1,58 @@
+@keyframes show_details_button {
```
Comment (ecino) [inline sms_sponsorship/static/src/css/custom.css]: Remove all classes that are not used in your template. You basically just need donation_success, thank_you_container and quote.

**Review by ecino — CHANGES_REQUESTED** [review]
Your css file should be loaded in the frontend assets of Odoo, otherwise it is ignored. Check how it's done in the muskathlon module.

File: sms_sponsorship/__manifest__.py [hunk 9]
```diff
@@ -24,7 +24,8 @@
         'views/event_compassion_view.xml',
         'views/hold_view.xml',
         'views/sms_child_request_view.xml',
-        'views/notification_settings_view.xml'
+        'views/notification_settings_view.xml',
+        'templates/sms_registration_confirmation_template.xml'
```
Comment (ecino) [inline sms_sponsorship/__manifest__.py]: Add sms_sponsorship/templates/assets.xml

**Review by ecino — CHANGES_REQUESTED** [review]
Almost ;)

**Comment by quentingigon** [comment]
I think the modification in file sms_child_request.py in my last commit fixes the issue of sponsorship validation

File: sms_sponsorship/templates/sms_registration_confirmation_template.xml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+            <t t-set="title">Thank you for registration!</t>
+            <div class="oe_structure donation_success">
+                <div class="thank_you_container">
+                    <div class="thank_you_image_container">
+                        <img class="img-responsive" style="width: 100%" src="/sms_sponsorship/static/src/img/compassion_thankyou_donation.jpg"/>
+                    </div>
+                    <div class="thank_you_text_container">
+                        <h1>Registration successful !</h1>
+                        <p class="quote">
+                            Thank you for your interest for your registration !
```
Comment (ecino) [inline sms_sponsorship/templates/sms_registration_confirmation_template.xml]: This text is not right. I would put `Let a new story for <t t-esc="sponsorship.child_id.preferred_name"/> begin! You will soon receive all documentation needed to start the sponsorship in your mailbox.`

File: sms_sponsorship/forms/sms_sponsorship_registration_form.py [hunk 11]
```diff
@@ -184,8 +184,10 @@ def form_after_create_or_update(self, values, extra_values):
             # validate sponsorship and send confirmation email
             sponsorship = self.main_object.sudo()
             sms_request = self.env['sms.child.request'].sudo().search([
-                ('sponsorship_id', '=', self.main_object.id)
+                ('partner_id', '=', sponsorship.partner_id.id)
```
Comment (ecino) [inline sms_sponsorship/forms/sms_sponsorship_registration_form.py]: I'm not sure this is a good idea. Looking for the sponsorship id can only lead to the correct request while a partner may make several requests. I would keep sponsorship_id here.

File: sms_sponsorship/models/sms_child_request.py [hunk 12]
```diff
@@ -66,7 +66,7 @@ class SmsChildRequest(models.Model):
 
     new_partner = fields.Boolean('New partner ?',
                                  help="is true if partner was created when "
-                                 "sending sms", default=True)
+                                 "sending sms", default=False)
```
Comment (ecino) [inline sms_sponsorship/models/sms_child_request.py:69]: If this fixes the bug, you should have a commit named CO-1965.

**Review by ecino — CHANGES_REQUESTED** [review]

File: sms_sponsorship/models/sms_child_request.py (hunk 12 above)
Comment (quentingigon) [reply sms_sponsorship/models/sms_child_request.py:69]: Ok, on this pull request or on a new one ?

Comment (ecino) [reply sms_sponsorship/models/sms_child_request.py:69]: On this one directly. I'm waiting for your fixes.

Comment (ecino) [reply sms_sponsorship/models/sms_child_request.py:69]: You can amend commit to edit the message just in case.

**Review by ecino — APPROVED** [review]

**Pull request merged**